lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -213,7 +237,7 @@ def run_sync(command, stdin = nil)
           started = true
         end
       end
-      session.process(0) until started
+      session.loop(0.1) { !started || (use_sudo_password? && !password_sent) }
```

Comment:
When testing, I got the script stuck here, because some commands were actually not using `sudo` (the run_sync is in rex used mostly for preparing the execution environment, while the sudo is being used just to run the actual user script. 

What was the reason to actually change the original line here (as well as in `run_async` method?). It seems that when I reverted this particular changes, the things started working.

Reply:
I wasn't sure when run_sync is used and mirrored the changes in run_async. I suppose calling ```session.process``` in a loop is sufficient, but I'm curious now why ```session.loop``` gets stuck, as it is *almost* the same.

Reply:
I think it's not about the `loop, but about the actual condition `(use_sudo_password? && !password_sent)` as when `sudo` is not used in the command, it would never be true.

OT: why do you prefer `loop(0.1)` over `process(0)`? Performance?

Reply:
> I think it's not about the loop, but about the actual condition(use_sudo_password? && !password_sent)as whensudo` is not used in the command, it would never be true.

Hmm. The whole expression ```!started || (use_sudo_password? && !password_sent)``` will evaluate to false (which is when the loop is terminated) once the ```started``` flag switched to ```true``` though?

> OT: why do you prefer loop(0.1) over process(0)? Performance?

It looks to me that loop makes calls ```process(0)``` one more time after the loop terminates. I'm not sure about the exact reasoning for this, but my guess is that eliminates/lowers the chance of losing incoming/outgoing data (which is only received from/sent onto the wire when ```process``` is called).